pkg/subctl/cmd/show_connections.go

```diff
@@ -72,14 +63,19 @@ func showConnections(cmd *cobra.Command, args []string) {
 	for _, item := range configs {
 		fmt.Println()
 		fmt.Printf("Showing information for cluster %q:\n", item.clusterName)
-		status := getConnectionsStatus(item.config)
-		printConnections(status)
+		showConnectionsFor(GetSubmarinerResource(item.config))
 	}
 }
 
 func showConnectionsFromConfig(config *rest.Config) {
-	status := getConnectionsStatus(config)
-	printConnections(status)
+	showConnectionsFor(GetSubmarinerResource(config))
+}
+
+func showConnectionsFor(submariner *v1alpha1.Submariner) {
+	if submariner != nil {
```

Comment:
It can be nil when called from L66 above.

Reply:
Sure, in that case I'd expect an error so that a bug can be opened, instead of a silent failure

Reply:
No - nil means submariner isn't installed.

Reply:
`showConnectionsFor` (or any other such function) is a private function and is not expected to be called with a `nil` value.
The value checking should be done outside or it can hide bugs as I already explained.

Reply:
Well there are no private functions in Go - it's package scoped. The API contract for `showConnectionsFor` is what ever one wants it to be. It doesn't really matter who checks for nil in this case - if you feel strongly about it outside then that's fine.

Reply:
I slightly agree with @mkolesnik here, if we are already checking for nil... it's not really necessary to check twice here... we don't expect this to be called with nil ever.

Anyway it will also work with nil here also, just being silent.

Reply:
Oh but wait, we can call those independently (not with the show all..)

I think it's good as it is then, avoids a lot of duplication around.

Reply:
> Oh but wait, we can call those independently (not with the show all..)
> 
> I think it's good as it is then, avoids a lot of duplication around.

On the contrary, this is exactly duplication of this exact line all over the place.
If not nil check is really needed (which it isnt), it can happen on line 80

Reply:
Right, ok, finally I settle on... handling that nil (and printing) outside.

For show_all.go it would only happen once ...

then for show_xxx.go it needs to happen on each of them before attempting to pass down the submariner object. Then... the nil checks on each function can be removed.

Reply:
"_...(and printing) outside._". I assume you mean printing of "_Submariner is not installed_"?  So you want to move that out of `getSubmarinerResource` and duplicate it 4-5 times?

Reply:
if status == nil { 
    print submariner not installed
} else {
    printConnections(status)
}